### ISSUE TITLE
Web UI - JS Error / Not Working

### ISSUE THREAD

**Issue by artdude543** [rethinkdb/rethinkdb#5482]
Hello,

I recently changed my ReThinkDB setup to use a Proxy for the apps to connect too. We followed the docs to the point and it all seems to be working; but the web UI seems to be broken. (I'm assuming we connect to the web app via the proxy IP (as stated in the docs)) but this bug also affects the other cluster instances running on the server.

This is the error being thrown in the console.
![xp4l-ed06f-1457086481](https://cloud.githubusercontent.com/assets/1885466/13524243/f83e2aca-e1f1-11e5-8898-3ee5b6c04596.png)

I believe all the configs are correct as the database is running fine, just the panel pretty much useless for some reason.

We are running version: `2.2.5` and Ubuntu `14.04` on the connected servers. (All running the same RethinkDB version) The data explorer works fine however just the rest of the pages / stats not working due to some odd error. (Not sure if the console error is related but it started when we moved to the proxy setup)

If you need more information about the setup please do ask. Really looking to find out if this is something I'm doing wrong or a bug with the panel.

Thanks,
Kieron

**Comment by danielmewes** [comment]
Hi @artdude543, It almost looks like this error is unrelated to the proxying.

Can you still open the web UI? If yes, are you able to navigate to the Data Explorer, and run the following query?
```js
r.db('rethinkdb').table('table_config')
```

If this works, could you post the query output here?

**Comment by danielmewes** [comment]
Oh and could you run
```js
r.db('rethinkdb').table('server_config')
```
and
```js
r.db('rethinkdb').table('current_issues')
```

as well?

**Comment by artdude543** [comment]
Sure can do! Will get the outputs for you now. (Sorry for the late reply was at work)

**Comment by artdude543** [comment]
Yeah the data explorer works fine. Just the other pages don't work. I.E. The `Tables` page never loads the databases / tables listing. The Server listing takes around 5/10mins to show anything and then more time to view details about a server. But the main page shows the active graph working fine but the stats take a while to load and sometimes don't load at all.

Here are the outputs as requested.

- Table Config: [Output](https://gist.github.com/artdude543/ef7458c8cb86a3540b8c)
- Server Config: [Output](https://gist.github.com/artdude543/2215a1d8de573d835eb1)
- Current Issues: `[]` (No results found)

**Comment by danielmewes** [comment]
Thanks for the configuration data @artdude543 .

It looks like the query that's failing is this one:
```coffeescript
fetch_data: (server_uuid) =>
        query = r.expr
            tables: r.db(system_db).table('table_config').pluck('db', 'name', 'id').coerceTo("ARRAY")
            servers: r.db(system_db).table('server_config').pluck('name', 'id').coerceTo("ARRAY")
            issues: driver.queries.issues_with_ids()
            num_issues: r.db(system_db).table('current_issues').count()
            num_servers: r.db(system_db).table('server_config').count()
            num_tables: r.db(system_db).table('table_config').count()
            num_available_tables: r.db(system_db).table('table_status')('status').filter( (status) ->
                status("all_replicas_ready")
            ).count()
            me: r.db(system_db).table('server_status').get(server_uuid)('name')
```

The `table_config` and `server_config` outputs you posted look fine to me, which leaves `table_status` and `server_status` as potential sources of unexpected data.

A third option is that `/ajax/me` returns something odd.

@artdude543 Could you try three more things and post the results here?
```js
r.db('rethinkdb').table('table_status')
```
```js
r.db('rethinkdb').table('server_status')
```
and finally go to `/ajax/me` on the RethinkDB server in your browser (e.g. if you're accessing the web UI through `http://server:8080`, go to `http://server:8080/ajax/me`)? Usually this should give you the UUID of the server you're connecting to.

Thanks for your help in tracking this down!

**Comment by artdude543** [comment]
No worries happy to help out with finding out on what's going on.

So the outputs as request.

- Table Status: [Output](https://gist.github.com/artdude543/85d06a2814e1812dcc28)
- Server Status: [Output](https://gist.github.com/artdude543/dc2cb20eea38e4ba0fae)
- /ajax/me: `"8d71875b-bace-4a3a-a176-11c26107aee0"`

I stared out the IP's from the listing (I don't think they make a difference in the output for you) But if you need more debug data one can provide :)

**Comment by danielmewes** [comment]
Thanks for the outputs @artdude543 .

This was helpful. The string from `ajax/me` is usually the ID of a server, which should be appearing in `server_status`. However proxies do not appear in `server_status`, and that breaks the web UI.

I can actually reproduce this now. I'm not quite sure why we haven't noticed this issue before.

For what it's worth, this is working fine on `next`.

We'll try try to fix this for the next RethinkDB bug fix release (2.2.6). The work-around is to connect the web UI to one of the non-proxy servers, where it should work fine.

**Comment by artdude543** [comment]
We have a panel running on one of the clusters. But the panel does not work either, only difference is the `ajax/me` request is returning the right server ID for the server I'm connected too.

**Comment by artdude543** [comment]
But thank you for looking into this issue means a lot to see awesome people helping others and the support here is amazing ^_^

**Comment by danielmewes** [comment]
That's interesting. For the non-proxy panel, are you connecting directly to the server or is it through some sort of HTTP proxy or other indirection? Which browser is this with?

**Comment by artdude543** [comment]
It's both broken via an NGINX proxy and connecting direct via the IP/Port.

**Comment by danielmewes** [comment]
504 seems to be "Gateway timeout" http://www.checkupdown.com/status/E504.html .

Are you getting the same error when connecting directly (not via NGINX)? Or is that failing with a different error?

**Comment by artdude543** [comment]
Same error when connecting via both methods.

**Comment by danielmewes** [comment]
This is probably fixed in RethinkDB 2.3.0, since we changed how server IDs are identified.

@artdude543 I'm going to close the issue. Please let us know if it still persists for you on RethinkDB 2.3.0.
